plugins/net_plugin/net_plugin.cpp

```diff
@@ -975,9 +1000,22 @@ namespace eosio {
          close();
          return false;
       } else {
-         fc_dlog( logger, "connected to ${peer}", ("peer", peer_name()) );
-         socket_open = true;
-         start_read_message();
+
+         auto start_read = [this](){
+            fc_dlog( logger, "connected to ${peer}", ("peer", peer_name()) );
+            socket_open = true;
+            start_read_message();
+         };
+         if (use_ssl) {
+            ssl_socket->async_handshake(ssl::stream_base::server,
+                                       [start_read](const auto& ec){
+                                          EOS_ASSERT(!ec, ssl_handshake_error, "ssl handshake error: ${e}", ("e", ec.message()));
+                                          start_read();
+                                       });
```

Comment:
You need to `bind_executor` to the `strand` the lambda. See for example `async_connect`.